server/src/main/java/io/druid/client/CacheUtil.java

```diff
@@ -59,25 +59,11 @@
 
   public static void populate(Cache cache, ObjectMapper mapper, Cache.NamedKey key, Iterable<Object> results)
   {
+    List<Object> resultList = Lists.newArrayList(results);
```

Comment:
Please include either a unit test or a comment explaining why materializing the list is important.

Btw, I think we could avoid materializing the list by creating a JsonGenerator from the mapper, and using that generator to write the objects one by one to a ByteArrayOutputStream. (I think the JsonGenerator tracks enough state to avoid the problem you mentioned.) I'm not sure how much that matters though.

Reply:
The JsonGenerator approach may be better anyway, since it's compatible with old cached data (important for remote caches like memcached, that persist through Druid server restarts). Otherwise Druid servers will blow up when they expect a JSON array but instead see the start of an object.